File: app/src/main/java/br/com/mov/views/fragment/SearchFragment.java
package br.com.mov.views.fragment;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.view.View;
import android.view.ViewGroup;
import android.widget.SearchView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;
import java.util.List;

import br.com.mov.R;
import br.com.mov.models.Movie;
import br.com.mov.views.recyclerview.adapter.MovieAdapter;

public class SearchFragment extends Fragment {

    private MovieAdapter adapter;
    private RecyclerView movieList;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_search_layout, container, false);
        getActivity().setTitle("Search");

        movieList = view.findViewById(R.id.Rv_search);

        List<Movie> movies = new ArrayList<>();
        movies.add(new Movie("Moana", R.drawable.moana));
        movies.add(new Movie("Black P", R.drawable.blackp));
        movies.add(new Movie("The Martian", R.drawable.themartian));
        movies.add(new Movie("Moana", R.drawable.moana));
        movies.add(new Movie("Black P", R.drawable.blackp));
        movies.add(new Movie("The Martian", R.drawable.themartian));
        movies.add(new Movie("Moana", R.drawable.moana));
        movies.add(new Movie("Black P", R.drawable.blackp));
        movies.add(new Movie("The Martian", R.drawable.themartian));
        movies.add(new Movie("Moana", R.drawable.moana));
        movies.add(new Movie("Black P", R.drawable.blackp));
        movies.add(new Movie("The Martian", R.drawable.themartian));

        this.adapter = new MovieAdapter(getContext(), movies);
        movieList.setAdapter(adapter);

        setHasOptionsMenu(true);

        return view;
    }

    @Override
    public void onCreateOptionsMenu(@NonNull Menu menu, @NonNull MenuInflater inflater) {
        inflater.inflate(R.menu.search_menu, menu);

        MenuItem searchItem = menu.findItem(R.id.action_search);
        SearchView searchView = (SearchView) searchItem.getActionView();

        searchView.setOnQueryTextListener(new SearchView.OnQueryTextListener() {
            @Override
            public boolean onQueryTextSubmit(String query) {
                return false;
            }

            @Override
            public boolean onQueryTextChange(String newText) {
                if (adapter != null)
                    adapter.getFilter().filter(newText);
                return false;
            }
        });

        super.onCreateOptionsMenu(menu, inflater);
    }
}

File: app/src/main/java/br/com/mov/views/fragment/SearchFragment.kt
package br.com.mov.views.fragment

import android.os.Bundle
import android.view.*
import android.widget.SearchView
import androidx.fragment.app.Fragment
import androidx.navigation.fragment.NavHostFragment
import br.com.mov.R
import br.com.mov.views.recyclerview.adapter.MovieAdapter
import br.com.mov.views.viewmodel.MovieViewModel
import br.com.mov.views.viewmodel.StateAppComponentsViewModel
import br.com.mov.views.viewmodel.VisualComponents
import kotlinx.android.synthetic.main.fragment_search_layout.*
import org.koin.android.viewmodel.ext.android.sharedViewModel

class SearchFragment : Fragment() {

    private var adapter: MovieAdapter? = null
    private val appComponentsViewModel: StateAppComponentsViewModel by sharedViewModel()
    private val movieViewModel: MovieViewModel by sharedViewModel()
    private val navController by lazy { NavHostFragment.findNavController(this) }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {

        appComponentsViewModel.havCoponent = VisualComponents(true)

        return inflater.inflate(R.layout.fragment_search_layout, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        initMovieAdapter()

        setHasOptionsMenu(true)
    }

    private fun initMovieAdapter() {
        movieViewModel.checkMoviesReturned()?.observe(viewLifecycleOwner,
                androidx.lifecycle.Observer { movieList ->
                    if (movieList != null) {
                        this.adapter = context?.let { context ->
                            MovieAdapter(context, movieList)
                        }!!
                        rv_search.adapter = adapter
                    }
                })
    }

    private fun goToMovieDetailFragment() {
        navController.navigate(R.id.action_searchFragment_to_movieDetailFragment)
    }

    override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
        inflater.inflate(R.menu.search_menu, menu)

        val searchItem = menu.findItem(R.id.action_search)
        val searchView = searchItem.actionView as SearchView

        searchView.setOnQueryTextListener(object : SearchView.OnQueryTextListener {
            override fun onQueryTextSubmit(query: String): Boolean {
                return false
            }

            override fun onQueryTextChange(newText: String): Boolean {
                if (adapter != null)
                    adapter?.filter?.filter(newText)
                return false
            }
        })

        super.onCreateOptionsMenu(menu, inflater)
    }
}
